### PR TITLE
support bytecheck-secured argument passing

### PR DESCRIPTION
**Pull request by miloszm**
support bytecheck-secured argument passing

implements #1371

### CODE REVIEW

**Review by ureeves — DISMISSED** [review]
LGTM!

**Review by ureeves — APPROVED** [review]

**Review by ureeves — APPROVED** [review]

File: rusk/tests/rusk-state.rs [hunk 1]
```diff
@@ -72,7 +72,12 @@ where
                 .expect("current commit should exist");
 
         session
-            .call::<_, Note>(TRANSFER_CONTRACT, "push_note", &note, u64::MAX)
```
[the remark this answers is not in the document]
Comment (ureeves) [reply rusk/tests/rusk-state.rs:75]: Because the arguments are indeed wrong

[the remark this answers is not in the document]
Comment (miloszm) [reply rusk/tests/rusk-state.rs:75]: arguments passed were incorrect, block height was missing

[the remark this answers is not in the document]
Comment (herr-seppia) [reply rusk/tests/rusk-state.rs:75]: then that should b changed on recovery-state too

**Review by herr-seppia — CHANGES_REQUESTED** [review]

File: rusk/tests/rusk-state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (herr-seppia) [reply rusk/tests/rusk-state.rs:75]: I was wrong, recovery use "mint" since a while now

**Review by herr-seppia — APPROVED** [review]

**Review by ureeves — APPROVED** [review]

**Pull request merged**